### PR TITLE
Add explainer.

### LINKED ISSUE #59: TAG review
Part of the process for shipping stuff is a review by the TAG.
Someone needs to file a ticket and fill in the questionnaires for that.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes https://github.com/w3c/mediacapture-screen-share/issues/59.

[Preview](https://github.com/w3c/mediacapture-screen-share/blob/afed76e52d6880fec6f23a6d9e83922bddb6c0fc/explainer.md)

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Screen Capture Explained
+
+## What is it? What’s it for?
+
+It’s an API to get a live video stream of your computer screen or a single window on it, optionally with sound.
+
+It’s sometimes referred to it as screen-sharing, even though it’s only the capturing part being defined (sharing is just the primary use case).
+
+Have you ever been a presenter in an online video conference call? You probably shared your screen or the window of a presentation-program to do that. Have you ever presented in an online video conference call without this feature? That would probably require printouts and hand-waving! It’s a pretty key feature. This is that feature.
```
Comment (martinthomson) [inline explainer.md]: If content is online, it is possible to have everyone look at their own copy of that content, but that can lead to confusion as everyone can get a subtly different view.

File: explainer.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+for (const track of stream.getTracks()) peerConnnection.addTrack(track, stream); // share
+
+await stream.getVideoTracks()[0].applyConstraints({height: 480}); // downscale
+
+for (const track of stream.getTracks()) track.stop(); // stop capture & relinquish access
+```
+
+Some important differences and unique properties of screen capture:
+
+ * The request is gated by end-user activation
```
Comment (martinthomson) [inline explainer.md]: Check list construction.  You have a full sentence on the second-to-last bullet, which suggest that all bullets should be full sentences.

File: explainer.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Have you ever been a presenter in an online video conference call? You probably shared your screen or the window of a presentation-program to do that. Have you ever presented in an online video conference call without this feature? That would probably require printouts and hand-waving! It’s a pretty key feature. This is that feature.
+
+
+ * Not the networking part—that’s WebRTC’s RTCPeerConnection
+ * Not the camera and microphone part—that’s MediaCapture’s getUserMedia
+ * **Just** the part that captures your computer's display. It’s called: getDisplayMedia
+
+You can capture the screen for other reasons as well, but this is the driving use-case.
+
+The approach is heavily modeled on our camera API, hence the name similarity. There’s a reason for that: both require user permission, and return a MediaStream object with a pair of highly privacy sensitive video and audio tracks—It’s just your screen instead of your face. The resulting MediaStream is a drop-in anywhere a MediaStream is accepted in the platform. Having obtained such a MediaStream, a website can trivially:
```
Comment (martinthomson) [inline explainer.md]: link to the camera API

File: explainer.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+## What is it? What’s it for?
+
+It’s an API to get a live video stream of your computer screen or a single window on it, optionally with sound.
+
+It’s sometimes referred to it as screen-sharing, even though it’s only the capturing part being defined (sharing is just the primary use case).
+
+Have you ever been a presenter in an online video conference call? You probably shared your screen or the window of a presentation-program to do that. Have you ever presented in an online video conference call without this feature? That would probably require printouts and hand-waving! It’s a pretty key feature. This is that feature.
+
+
+ * Not the networking part—that’s WebRTC’s RTCPeerConnection
```
Comment (martinthomson) [inline explainer.md]: link for WebRTC (generally, I think that it's a good idea to link to these major components; I don't know if the spec or the repo or the explainer is best, but I'll let you decide)

File: explainer.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ * Not the camera and microphone part—that’s MediaCapture’s getUserMedia
```
Comment (martinthomson) [inline explainer.md]: link

File: explainer.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+ * **Just** the part that captures your computer's display. It’s called: getDisplayMedia
+
+You can capture the screen for other reasons as well, but this is the driving use-case.
+
+The approach is heavily modeled on our camera API, hence the name similarity. There’s a reason for that: both require user permission, and return a MediaStream object with a pair of highly privacy sensitive video and audio tracks—It’s just your screen instead of your face. The resulting MediaStream is a drop-in anywhere a MediaStream is accepted in the platform. Having obtained such a MediaStream, a website can trivially:
+
+ * play the live stream of your screen locally in a video element,
+ * send it over a peer connection to another client,
+ * send it to a MediaRecorder, perhaps for subsequent upload to a server,
```
Comment (martinthomson) [inline explainer.md]: link

File: explainer.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+ * or all of the above (at once even).
+
+This flexibility is thanks to the MediaCapture-main spec which already did the heavy lifting of:
```
Comment (martinthomson) [inline explainer.md:24]: link to spec

File: explainer.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+ * The request is gated by end-user activation
+ * The end-user is always prompted each time with a picker to select what to share
+ * It is not possible to persist permission grants (only blocks)
+ * Application-supplied constraints are applied after selection, and do not narrow the end-user’s choices (unlike with camera)
+ * All choices are videos. Any accompanying audio is optional and requires consent 
+ * Screen capture sources are never exposed in enumerateDevices()
+
+## Security and privacy implications
+
+Allowing a web page to capture and effectively record an end-user’s screen carries significant privacy and security risks. There are three of them, in increasing order of severity:
```
Comment (martinthomson) [inline explainer.md]: "an end-user's" or "your"?

**Review by martinthomson — APPROVED** [review]
Thanks for writing this up @jan-ivar.  It seems fairly comprehensive and accessible.  I only wish the spec were.

**Review by henbos — APPROVED** [review]

**Pull request merged**